src/models/build.d.ts

```diff
@@ -0,0 +1,213 @@
+  avatar_url: string;
+  gravatar_id: string | null;
+  url: string;
+  html_url: string;
+  followers_url: string;
+  following_url: string;
+  gists_url: string;
+  starred_url: string;
+  subscriptions_url: string;
+  organizations_url: string;
+  repos_url: string;
+  events_url: string;
+  received_events_url: string;
+  type: string;
+  site_admin: boolean;
+  starred_at: string;
+}
+
+export interface SimpleCommit {
+  id: string;
+  tree_id: string;
+  message: string;
+  timestamp: string;
+  author: {
+    name: string;
+    email: string;
+  } | null;
+  committer: {
+    name: string;
+    email: string;
+  } | null;
+}
+
+export interface MinimalRepository {
+  id: number;
+  node_id: string;
+  name: string;
+  full_name: string;
+  owner: SimpleUser;
+  private: boolean;
+  html_url: string;
+  description: string | null;
+  fork: boolean;
+  url: string;
+  archive_url: string;
+  assignees_url: string;
+  blobs_url: string;
+  branches_url: string;
+  collaborators_url: string;
+  comments_url: string;
+  commits_url: string;
+  compare_url: string;
+  contents_url: string;
+  contributors_url: string;
+  deployments_url: string;
+  downloads_url: string;
+  events_url: string;
+  forks_url: string;
+  git_commits_url: string;
+  git_refs_url: string;
+  git_tags_url: string;
+  git_url: string;
+  issue_comment_url: string;
+  issue_events_url: string;
+  issues_url: string;
+  keys_url: string;
+  labels_url: string;
+  languages_url: string;
+  merges_url: string;
+  milestones_url: string;
+  notifications_url: string;
+  pulls_url: string;
+  releases_url: string;
+  ssh_url: string;
+  stargazers_url: string;
+  statuses_url: string;
+  subscribers_url: string;
+  subscription_url: string;
+  tags_url: string;
+  teams_url: string;
+  trees_url: string;
+  clone_url: string;
+  mirror_url: string | null;
+  hooks_url: string;
+  svn_url: string;
+  homepage: string | null;
+  language: string | null;
+  forks_count: number;
+  stargazers_count: number;
+  watchers_count: number;
+  size: number;
+  default_branch: string;
+  open_issues_count: number;
+  is_template: boolean;
+  topics: string[];
+  has_issues: boolean;
+  has_projects: boolean;
+  has_wiki: boolean;
+  has_pages: boolean;
+  has_downloads: boolean;
+  has_discussions: boolean;
+  archived: boolean;
+  disabled: boolean;
+  visibility: string;
+  pushed_at: string | null;
+  created_at: string | null;
+  updated_at: string | null;
+  permissions: {
+    admin: boolean;
+    maintain: boolean;
+    push: boolean;
+    triage: boolean;
+    pull: boolean;
+  };
+  role_name: string;
+  temp_clone_token: string;
+  delete_branch_on_merge: boolean;
+  subscribers_count: number;
+  network_count: number;
+  code_of_conduct: CodeOfConduct;
+  license: License | null;
+  forks: number;
+  open_issues: number;
+  watchers: number;
+  allow_forking: boolean;
+  web_commit_signoff_required: boolean;
+  security_and_analysis: SecurityAndAnalysis | null;
+}
```

Comment:
For all the values that can be a specified type or null, consider having them as optional, for example: `created_at?: string;`. This is just a suggestion and is subject to your preference.

Reply:
Aww no, someone else told me differently, I do not know which is better

Reply:
I am not sure either, I think it is up to your preference. Just think if you do it one way, to stick to it for consistency purposes.